### PR TITLE
Fix EnvStep comparison and tests

### PR DESCRIPTION
**Pull request by AiRuiChen**
1. Add EnvSpec Comparison logic so it works with `roundtrip.spec == env.spec` evaluation.
2. Fix GymEnv's get time limit logic - `gym.TimeLimit` doesn't have `spec` populated, so in this case we should look at `_max_episode_steps` directly.
3. Fix some example usages.

Closes [#1896 ](https://github.com/rlworkgroup/garage/issues/1896)

### CODE REVIEW

File: tests/helpers.py [hunk 1]
```diff
@@ -45,8 +44,15 @@ def step_env_with_gym_quirks(env,
     """
     if serialize_env:
         # Roundtrip serialization
-        round_trip = pickle.loads(pickle.dumps(env))
-        assert round_trip.env.spec == env.env.spec
+        round_trip = cloudpickle.loads(cloudpickle.dumps(env))
+        # print (round_trip.spec.observation_space,
```
Comment (yeukfu) [inline tests/helpers.py:48]: should remove?

File: src/garage/_environment.py [hunk 2]
```diff
@@ -74,6 +74,20 @@ def max_episode_length(self):
...
         return self._max_episode_length
 
+    def __eq__(self, other):
+        """See object.__eq__.
+
+        Args:
+            other (EnvSpec): the EnvSpec to compare with.
+
+        Returns:
+            True: if the attributes of EnvSpec are the same.
```
Comment (ryanjulian) [inline src/garage/_environment.py:84]:
```suggestion
            bool: Whether these :class:`~EnvSpec` instances are equal.
```

File: src/garage/_environment.py [hunk 3]
```diff
@@ -74,6 +74,20 @@ def max_episode_length(self):
         """
         return self._max_episode_length
 
+    def __eq__(self, other):
+        """See object.__eq__.
```
Comment (ryanjulian) [inline src/garage/_environment.py:78]:
```suggestion
        """See :meth:`object.__eq__`.
```

File: src/garage/_environment.py [hunk 4]
```diff
@@ -74,6 +74,20 @@ def max_episode_length(self):
...
+
+        Args:
+            other (EnvSpec): the EnvSpec to compare with.
```
Comment (ryanjulian) [inline src/garage/_environment.py:81]:
```suggestion
            other (EnvSpec): :class:`~EnvSpec` to compare with.
```

File: src/garage/_environment.py [hunk 5]
```diff
@@ -74,6 +74,20 @@ def max_episode_length(self):
...
+
+        Returns:
+            True: if the attributes of EnvSpec are the same.
+        """
+        return ((self.observation_space, self.action_space,
+                 self.max_episode_length) == (other.observation_space,
+                                              other.action_space,
+                                              other.max_episode_length))
```
Comment (ryanjulian) [inline src/garage/_environment.py:89]:
```suggestion
        return (
            self.observation_space == other.observation_space and
            self.action_space == other.action_space and
            self.max_episode_length == other.max_episode_length
        )
```

I'm surprised this isn't handled by namedtuple

File: tests/helpers.py (hunk 1 above)
Comment (ryanjulian) [reply tests/helpers.py:48]: 👍

Comment (AiRuiChen) [reply tests/helpers.py:48]: good catch!

File: src/garage/_environment.py (hunk 5 above)
Comment (AiRuiChen) [reply src/garage/_environment.py:89]: i think the default `__eq__` checks if they are the exact same instance (via looking at the address?). `==` wouldn't return true for `EnvSpec` with same values without this magic function change.

**Comment by AiRuiChen** [comment]
> Does this fix #1896?

It's passing the failed tests locally on my end.

**Pull request merged**